samples/gibbs_ensemble_client.py

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (C) 2013-2019 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Client part of the Gibbs ensemble simulation. This script handles the
+simulation boxes and communicates the energies to the host. The Monte-Carlo
+part of the simulation is done by the `gibbs_ensemble_socket.py` script.
+"""
+
+import espressomd
+import socket
+import numpy as np
+import pickle
+import struct
+import argparse
+
+seed = None
+init_box_l = None
+particle_number = None
+
+lj_epsilon = None
+lj_sigma = None
+lj_cutoff = None
+lj_shift = None
+
+HOST = 'localhost'
+PORT = 31415
+
+# Message identifiers
+MSG_START = 0
+MSG_END = 1 
+MSG_MOVE_PART = 2
+MSG_MOVE_PART_REVERT = 21
+MSG_CHANGE_VOLUME = 3
+MSG_EXCHANGE_PART_ADD = 4
+MSG_EXCHANGE_PART_ADD_REVERT = 41
+MSG_EXCHANGE_PART_REMOVE = 5
+MSG_EXCHANGE_PART_REMOVE_REVERT = 51
+MSG_ENEGRY = 6
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-n', '--number-particles', type=int, nargs=1)
+parser.add_argument('-lj', '--lj-params', type=float, nargs=4)
+parser.add_argument('-s', '--seed', type=int, nargs=1)
+parser.add_argument('-bl', '--box-length', type=float, nargs=1)
+
+args = parser.parse_args()
+
+if args.number_particles:
+    particle_number = args.number_particles[0]
+if args.lj_params:
+    lj_epsilon = args.lj_params[0]
+    lj_sigma = args.lj_params[1]
+    lj_cutoff = args.lj_params[2]
+    lj_shift = args.lj_params[3]
+if args.seed:
+    seed = args.seed[0]
+if args.box_length:
+    init_box_l = args.box_length[0]
+
+
+class Manipulator():
+    '''
+    Manipulator class, which has all the system changeing functions.
+    '''
+
+    def __init__(self, system):
+        self.m_system = system
```

Comment:
I think "private" variables are normally not prefixed with ```m_``` in python. Instead you can use name-mangling by using to underscores as a prefix, *e.g.* ```self.__system = system```

Reply:
Shouldn't name mangling (two leading underscores) only be used in the context of inheritance? [PEP8: Method Names and Instance Variables](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables). For "private" class attributes and methods, it is quite common to use the weak internal use indicator (a single leading underscore).

Reply:
https://docs.python.org/3/tutorial/classes.html#private-variables

Reply:
it's probably both ok